.github/workflows/test-docs.yml

```diff
@@ -72,3 +77,29 @@ jobs:
         run: |
           cd docs
           make linkcheck
+
+      # make rediraffecheckdiff compares files with the origin/main and other
+      # git references, and forks can be outdated, we only run these steps when
+      # working against jupyterhub/jupyterhub to avoid incorrectly erroring
+      - name: check redirects for this PR
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          cd docs
+          export REDIRAFFE_BRANCH=${{ github.base_ref }}
+          make rediraffecheckdiff
+
+      # this should check currently published 'stable' links for redirects
+      - name: check redirects since last release
+        if: github.repository == 'jupyterhub/jupyterhub'
+        run: |
+          cd docs
+          export REDIRAFFE_BRANCH=$(git describe --tags --abbrev=0)
+          make rediraffecheckdiff
+
+      # longer-term redirect check (fixed version) for older links
+      - name: check redirects since 3.0.0
+        if: github.repository == 'jupyterhub/jupyterhub'
```

Comment:
@minrk the `${{ }}` part isn't needed for `if` statements, but will be fine to have.

I added two more `github.repository == 'jupyterhub/jupyterhub'` checks below that needs to be updated as well. Maybe they aren't critical? Or are they?

I think maybe we will end up failing now when we check against `main`, but I'm not confident. I figure maybe we just checkout a branch or similar, and `main` isn't a checked out local branch, so we would be forced to compare with `origin/main` - or?

Hmmm... This is quite tricky to understand I think.

Reply:
I think we should lean towards having `if` conditions for all three, not just the first. And if we check against `main` by default in conf.py, maybe we need to do `export REDIRAFFE_BRANCH=origin/main` in the first step in this CI job.

Reply:
Yes, I think if for all makes sense - for the pr it resolves the ref regardless of origin (this run makes no sense for non PRs) and the other two because forks don't always have up-to-date tags.

I think "main" is the right default in the file (as it was before) because that will be used only by local runs, where we don't want to assume how people manage their remotes.

Reply:
> I think maybe we will end up failing now when we check against main, but I'm not confident.

To be clear: we won't ever check against main implicitly on CI. It's always a tag or the explicit base ref (which is usually main, but always guaranteed to exist and be correct when that is the case).